### PR TITLE
Also find PRs from check_suite.

### LINKED ISSUE #20: Get the PR from check_suite and check_runs payload
Currently, PyDocTeur won't be able to detect the PR from a check_run or check_suite completed payload.

This is because of the github API:

> Note: The Checks API only looks for pushes in the repository where the check suite or check run were created. Pushes to a branch in a forked repository are not detected and return an empty pull_requests array and a null value for head_branch.

https://developer.github.com/v3/checks/suites/

### PR DESCRIPTION
**Pull request by JulienPalard**
Closes #20

### CODE REVIEW

File: pydocteur/github_api.py [hunk 1]
```diff
@@ -24,37 +23,31 @@ def get_graphql_api(query: str) -> requests.Response:
...
+
 def get_pull_request(payload):
-    logger.debug("Getting repository")
     gh_repo = gh.get_repo(REPOSITORY_NAME)
-    logger.info("Trying to find PR number from payload")
+    logger.info("Trying to find PR from %s", payload.get("action", "A payload with: " + ", ".join(payload.keys())))
 
-    is_run = payload.get("check_run", False)
-    is_suite = payload.get("check_suite", False)
+    head_sha = payload.get("check_run", {}).get("head_sha")
```
Comment (Seluj78) [inline pydocteur/github_api.py]: You're only getting it from check_runs, why not check_suite ?

Comment (JulienPalard) [reply pydocteur/github_api.py]: Is it usefull?

Comment (JulienPalard) [reply pydocteur/github_api.py]: Or we should do the other way around: we should have less check_suite than check_run, but it would be enough?

Comment (JulienPalard) [reply pydocteur/github_api.py]: I also dropped check_run from the hooks, I bet we don't need to receive them.

File: pydocteur/github_api.py [hunk 2]
```diff
@@ -9,7 +9,7 @@
 from pydocteur.settings import REPOSITORY_NAME
 
 logger = logging.getLogger("pydocteur")
-gh = Github(GH_TOKEN)
+gh = Github(GH_TOKEN if GH_TOKEN else None)
```
Comment (Seluj78) [inline pydocteur/github_api.py:12]: Not needed, if GH_TOKEN not here then the code won't run. see https://github.com/AFPy/PyDocTeur/blob/main/pydocteur/settings.py#L35-L39

**Review by Seluj78 — CHANGES_REQUESTED** [review]

File: pydocteur/github_api.py (hunk 2 above)
Comment (JulienPalard) [reply pydocteur/github_api.py:12]: The tests pass without a token, and I don't want a token in the tests. Should we word this in some way to make it explicit?

Comment (Seluj78) [reply pydocteur/github_api.py:12]: Just tested and this is what i got: https://wyz.fr/4F-1M

From adding https://wyz.fr/0I-NN

Comment (Seluj78) [reply pydocteur/github_api.py:12]: Because `get_rest_api` uses `HTTPBasicAuth` to bypass rate limiting

Comment (Seluj78) [reply pydocteur/github_api.py:12]: (This is tested on the `main` branch, not on yours BTW)

Comment (JulienPalard) [reply pydocteur/github_api.py:12]: I don't follow, how we would care about rate limiting in the tests? They don't hit the actal API (they the cassettes).

**Pull request merged**